### ISSUE TITLE
CPU profiler disabled on iOS due to persistent `lldb` connection

### ISSUE THREAD

**Issue by lukaszdebowski** [flutter/flutter#88466]
Hi 👋🏻 

I have an issue with the Devtools - after clicking `record` and then `stop` in the CPU profiler tab, the results are not loaded at all, I just see an empty page. Am I doing something wrong? I'm pretty sure I've used that before and it worked after clicking the stop button 🤔 

Video for reference:

https://user-images.githubusercontent.com/39137870/127768162-561b5345-3046-4613-9847-cd6afb12e92b.mp4

I am using the newest version of the dev tools: 2.4.0
The app is running on real device (iphone 12 mini, latest iOS) using `flutter run --profile`

**Comment by jmagman** [comment]
I know this isn't a DevTools workaround or do anything to help with this bug, but if you have an urgent need for CPU profiling you can run in Xcode's Instruments developer tools.
```
$ open ios/Runner.xcworkspace
```
In Xcode, Product > Profile
Select Time Profiler or Activity Monitor, or whatever template works best for your needs.  Start recording.  You should see the `flutter` symbols.  That may be too low level, depending on what you're doing.

**Comment by jmagman** [comment]
This is a problem in iOS 13+, not just iOS 14.  `lldb` has been attached on `flutter run` since https://github.com/flutter/flutter/pull/66399 available in Flutter 1.23.0-7.0.pre from September 2020.  So this isn't a recent regression, as far as we know.

More context in https://github.com/dart-lang/sdk/issues/28702.
> Debugging on Mac via `lldb` is very slow when profiler is enabled. Currently when using lldb one has to run VM with `--no-profiler` to get reasonable performance. This is most likely due to SIGPROF signals flooding lldb.

On the `flutter` tooling side, we can tell `lldb` to ignore `SIGPROF`.
1. Update ios-deploy artifact to include https://github.com/ios-control/ios-deploy/pull/472
2. In the tool add a custom-command to not pass `SIGPROF`:
https://github.com/flutter/flutter/blob/c37fe7c4fa09e27b05f6be2be6fcb5130887ccb1/packages/flutter_tools/lib/src/ios/ios_deploy.dart#L150
```diff
       '--debug',
+      '--custom-command',
+      'process handle SIGPROF -n false -p false -s false',
```

Note: this won't do anything to help users who launch their app from Xcode, only from `flutter run`.

Then on the VM side, [ThreadInterrupter::IsDebuggerAttached()](https://github.com/dart-lang/sdk/blob/6e28f8bb404b3dd81407743bd8c16e422ad2bd76/runtime/vm/thread_interrupter_macos.cc#L30) on macOS should always return false (or all that logic should be removed).

If the app is still slow after lldb is ignoring `SIGPROF`, there's some other problem going on which should be investigated.

**Comment by bkonyi** [comment]
Enabling logs with `log enable lldb default` shows that even when `lldb` has been told to ignore a specific signal it will always suspend to check if the signal should be handled at all. It seems like this behavior is core to how `lldb` processes signals so I don't think this is something we can fix by passing some sort of flag.

**Comment by jmagman** [comment]
I tried detaching after a `flutter run`, which detaches the debugger.  Then I ran `flutter attach` which doesn't use the debugger or `ios-deploy` at all, it just leverages the mDNS advertised .  

```
$ flutter run -d <device-id>  --profile
...
Flutter run key commands.
h List all available interactive commands.
c Clear the screen
q Quit (terminate the application on the device).
An Observatory debugger and profiler on iPhone 6s is available at: http://127.0.0.1:60737/_KAJHuwLQK8=/
The Flutter DevTools debugger and profiler on iPhone 6s is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:60737/_KAJHuwLQK8=/

[press "d"]
Application finished.
```
Note you can press "d" to detach even though it's not listed in that help text. The app should still running on the device.  Then `flutter attach` to the same device.
```
$ flutter attach -d <device-id>
```
That should give you a new DevTools link. I still was unable to load any CPU samples even though `lldb` wasn't attached--it looked the same as the original recording.

@bkonyi Can you try this flow with your build that ignores the debugger state on launch?

**Comment by cyanglaz** [comment]
@lukaszdebowski @ristiisa, @jonataslaw Does @jmagman's workaround unblock your use case?

**Comment by ristiisa** [comment]
> @lukaszdebowski @ristiisa, @jonataslaw Does @jmagman's workaround unblock your use case?

yes

**Comment by cyanglaz** [comment]
Lower priority to P3 as we have a workaround. I'll defer to @jmagman to see if there is a viable fix any time soon.

**Comment by zanderso** [comment]
Closing this in favor of https://github.com/dart-lang/sdk/issues/47139
